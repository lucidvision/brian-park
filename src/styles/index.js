export const colors = {
  primary: 'rgba(0, 13, 85, 0.8)',
  black: '#000',
  white: '#fff',
  gradientLight: 'rgba(0, 38, 255, 0.8)',
  gradientDark: 'rgba(0, 0, 0, 0.8)'
}

export const fontSizes = {
  large: '6rem',
  medium: '4rem',
  small: '2.2rem',
  default: '1.6rem'
}

export const screenSizes = {
  desktop: '112.5em',
  tablet: '75em',
  phone: '56.25em'
}

export const utilities = {
  marginSmall: '1.5rem',
  marginMedium: '4rem',
  marginLarge: '8rem'
}
