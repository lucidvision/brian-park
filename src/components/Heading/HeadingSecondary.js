import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors, fontSizes } from 'styles'

HeadingSecondary.propTypes = {
  color: PropTypes.string,
  text: PropTypes.string.isRequired
}

HeadingSecondary.defaultProps = {
  color: colors.primary,
  text: ''
}

const StyledHeading = styled.h1`
  color: ${props => props.color};
  font-size: ${fontSizes.medium};
  text-transform: uppercase;
  letter-spacing: .5rem;
`

export default function HeadingSecondary (props) {
  const { text, ...rest } = props
  return (
    <StyledHeading {...rest}>{props.text}</StyledHeading>
  )
}
