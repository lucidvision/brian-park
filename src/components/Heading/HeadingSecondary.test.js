import React from 'react'
import { mount } from 'enzyme'
import HeadingSecondary from './HeadingSecondary'
import { colors } from 'styles'

const createHeadingSecondary = (override) => {
  const props = {
    text: 'Unicorns',
    ...override
  }

  return mount(<HeadingSecondary {...props} />)
}

describe('HeadingSecondary', () => {
  const component = createHeadingSecondary()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders with the correct props', () => {
    expect(component.find('HeadingSecondary').props()).toEqual({
      color: colors.primary,
      text: 'Unicorns'
    })
  })

  it('renders h2 with the correct text', () => {
    expect(component.find('h2').props().children).toBe('Unicorns')
  })
})
