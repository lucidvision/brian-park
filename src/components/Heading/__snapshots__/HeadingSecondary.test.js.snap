// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingSecondary renders correctly 1`] = `
<HeadingSecondary
  color="rgba(0, 13, 85, 0.8)"
  text="Unicorns"
>
  <styled.h2
    color="rgba(0, 13, 85, 0.8)"
  >
    <h2
      className="sc-bdVaJa iBikEu"
      color="rgba(0, 13, 85, 0.8)"
    >
      Unicorns
    </h2>
  </styled.h2>
</HeadingSecondary>
`;
